src/lib.rs

```diff
@@ -1,163 +1 @@
-
-        assert_eq!(bucket.consume(), true);
-        assert_eq!(bucket.consume(), false);
-
-        *now.borrow_mut() += Duration::from_secs(1);
-        assert_eq!(bucket.consume(), true);
-        assert_eq!(bucket.consume(), false);
-    }
-
-    #[test]
-    fn capacity_gt_one() {
-        let now = Rc::new(RefCell::new(Instant::now()));
-        let now_moved = now.clone();
-
-        let mut bucket = TokenBucket::new(3, Duration::from_secs(1))
-            .with_timer(Box::new(move || *now_moved.borrow()));
-
-        assert_eq!(bucket.consume(), true);
-        assert_eq!(bucket.consume(), true);
-        assert_eq!(bucket.consume(), true);
-        assert_eq!(bucket.consume(), false);
-
-        *now.borrow_mut() += Duration::from_secs(1);
-        assert_eq!(bucket.consume(), true);
-        assert_eq!(bucket.consume(), true);
-        assert_eq!(bucket.consume(), true);
-        assert_eq!(bucket.consume(), false);
-    }
-
-    #[test]
-    fn period_gt_one() {
-        let now = Rc::new(RefCell::new(Instant::now()));
-        let now_moved = now.clone();
-
-        let mut bucket = TokenBucket::new(1, Duration::from_secs(3))
-            .with_timer(Box::new(move || *now_moved.borrow()));
-
-        assert_eq!(bucket.consume(), true);
-        assert_eq!(bucket.consume(), false);
-
-        *now.borrow_mut() += Duration::from_secs(3);
-        assert_eq!(bucket.consume(), true);
-        assert_eq!(bucket.consume(), false);
-    }
-
-    #[test]
-    fn consume_over_time() {
-        let t0 = Instant::now();
-        let now = Rc::new(RefCell::new(t0.clone()));
-        let now_moved = now.clone();
-
-        let mut bucket = TokenBucket::new(4, Duration::from_secs(1))
-            .with_timer(Box::new(move || *now_moved.borrow()));
-
-        // consume first token
-        *now.borrow_mut() = t0;
-        assert_eq!(bucket.consume(), true);
-
-        // consume second token
-        *now.borrow_mut() = t0 + Duration::from_millis(50);
-        assert_eq!(bucket.consume(), true);
-
-        // consume third & fourth tokens
-        *now.borrow_mut() = t0 + Duration::from_millis(150);
-        assert_eq!(bucket.consume(), true);
-        assert_eq!(bucket.consume(), true);
-
-        // ensure we are out of tokens
-        assert_eq!(bucket.consume(), false);
-
-        // one token is not yet replenished
-        *now.borrow_mut() = t0 + Duration::from_millis(249);
-        assert_eq!(bucket.consume(), false);
-
-        // one token is replenished
-        *now.borrow_mut() = t0 + Duration::from_millis(250);
-        assert_eq!(bucket.consume(), true);
-
-        // ensure we are out of tokens again
-        assert_eq!(bucket.consume(), false);
-
-        // two tokens are replenished
-        *now.borrow_mut() = t0 + Duration::from_millis(750);
-        assert_eq!(bucket.consume(), true);
-        assert_eq!(bucket.consume(), true);
-        assert_eq!(bucket.consume(), false);
-    }
-}
+pub mod token_bucket;
```

Comment:
What people frequently do in this case to avoid exposing internal module structure (or so that library users don't have to remember where the names should be imported from) is something like:

```rust
mod token_bucket;

pub use token_bucket::TokenBucket;
```

Up to you.